tests/test_cocos.py

```diff
@@ -75,3 +75,14 @@ def test_cocos_transform():
         for cocos_add in (cocos + x * 10 for x in range(2)):
             for key in ("b_toroidal", "toroidal", "poloidal", "q"):
                 assert getattr(Transform(cocos_add, cocos_add), key) == 1
+
+
+@pytest.mark.parametrize("bad_input", ["B_poloidal", "r_phi_z", "r_theta_phi"])
+def test_sigma_bad_inputs(bad_input):
+    """Test that Sigma raises an excpetion when given an inputs not in ``(-1, 1)``"""
+    inputs = {
+        k: (0 if k == bad_input else 1)
+        for k in ("B_poloidal", "r_phi_z", "r_theta_phi")
+    }
```

Comment:
This is fine, but maybe borderline too clever for this kind of test. I would perhaps just do the simpler, stupider test like:

```python
    with pytest.raises(ValueError, match="B_poloidal"):
        Sigma(B_poloidal=2, r_phi_z=1, r_theta_phi=-1)

    with pytest.raises(ValueError, match="r_phi_z"):
        Sigma(B_poloidal=1, r_phi_z=0, r_theta_phi=-1)
```

just for the explicitness

Reply:
Good point, I think I got caught up on the idea that it needed to be parametrized. You should have seen the version I wrote using `set(permutations(-1, 0, 1))`, even I realised that one was getting a bit much.